spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/core/Mqttv3ClientManager.java

```diff
@@ -198,4 +198,11 @@ public void deliveryComplete(IMqttDeliveryToken token) {
 		// nor this manager concern
 	}
 
+	@Override
+	public boolean isConnection() {
+		if (getClient() != null) {
+			return getClient().isConnected();
```

Comment:
Better to extract into a local variable.
I mean `IMqttAsyncClient client = getClient()`.
Also wrap the logic into a `this.lock.lock();`.
See `start()` for example.